src/plone/app/layout/globals/layout.py

```diff
@@ -295,6 +296,30 @@ def bodyClass(self, template, view):
 
         return " ".join(sorted(body_classes))
 
+    @memoize
+    def is_xhr(self):
+        """Check if the current request is an XHR request."""
+        return self.request.get("HTTP_X_REQUESTED_WITH") == "XMLHttpRequest"
+
+    @memoize
+    def use_ajax(self):
```

Comment:
Note: I did not change the `use_ajax` method here while the plone.base setting was changed to `use_ajax_main_template`.
I think `use_ajax` might be also useful in other contexts than switching than switching the main template.

But I also not 100% confident about this. If it's only about switching the main_template this method could also go directly into Products.CMFPlone.browser.main_template.

Since we want to get this right, I'd like a feedback on that.